### PR TITLE
Add support for loading mods in 1.13 subdirectory

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by UpcraftLP** [comment]
Could this pull the mc version instead of hardcoding it? (so it doesn't break in newer versions)

**Comment by gbui** [comment]
I think Rift should have a constants class in general that includes the Rift version as well as the Minecraft version. Currently, `1.13` is also hardcoded in the `Main` class.

File: src/main/java/org/dimdev/riftloader/RiftLoader.java [hunk 1]
```diff
@@ -127,7 +132,12 @@ private void findMods(File modsDir) {
                 log.error("Exception while checking if file " + file + " is a mod", t);
             }
         }
+    }
 
+    private void findMods() {
+        findModsOnClasspath();
+        findModsInDir(modsDir);
+        findModsInDir(new File(modsDir, "1.13"));
```
Comment (sjx233) [inline src/main/java/org/dimdev/riftloader/RiftLoader.java]: Why not use `RealmsSharedConstants.VERSION_STRING`?

> Currently, `1.13` is also hardcoded in the `Main` class.

It is inlined by the compiler, isn't it?

**Comment by Runemoro** [comment]
The problem with this is that Rift for 1.13 mods will still be loaded in 1.13.1, and possibly even 1.14 if there aren't too many breaking changes. People wouldn't know if they should place a 1.13 mod they want to load on 1.13.1 in the 1.13 folder (because it's a 1.13 mod) or the 1.13.1 folder (because they want it to be loaded in 1.13.1).

Determining whether a mod is outdated or not should be instead done with a field in `riftmod.json`. Mods that are outdated but for which compatibility with the new version is possible will be remapped, transformed, and loaded, while those that are too outdated will be ignored (with a warning message to the player about the mod being too outdated).

**Comment by ghost** [comment]
Makes sense. In that case, what about a folder/subfolder specifically for Rift mods? The main reason I PR'ed this was this (quoted from #rift-support on Discord):

> support for doing this with rift like you can do it with every forge version when
> 
> tbh dont want to put it in the main directory because forge will load it into the classpath on 1.12

**Comment by Runemoro** [comment]
> tbh dont want to put it in the main directory because forge will load it into the classpath on 1.12

Why not just set a different game directory? That way you make sure everything is separate, including the config folder.

Another option would be making the mods (and config) directory locations configurable, possibly through a command line option (which can be easily added in the Minecraft launcher).

**Comment by kierajreed** [comment]
I think mod subdirectories are good, but a command line option would be great too. That way all users are happy.

**Comment by UpcraftLP** [comment]
> Another option would be making the mods (and config) directory locations configurable, possibly through a command line option (which can be easily added in the Minecraft launcher).

Looks like we'll end up having a mod list format like the one forge introduced :P

**Pull request closed without merging**